### PR TITLE
Ft post service

### PR DESCRIPTION
**Pull request by MSmaili**
Created test for post

### CODE REVIEW

File: src/main/java/com/sda/microblogging/service/PostService.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        return postRepository.findAll();
+    }
+
+    public List<Post> findPostsByOwner(int userId) {
+        return postRepository.findPostsByOwner(userId);
+    }
+
+    public Post save(Post post) {
+
+        requireNonNull(post.getContent());//should have a content
```
Comment (Yarulika) [inline src/main/java/com/sda/microblogging/service/PostService.java:29]: If empty string is ok?

File: src/test/java/com/sda/microblogging/service/PostServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+        assertEquals(2,actualPosts.size());
+        verify(postRepository,times(1)).findAll();
+    }
+    @Test
+    public void get_all_posts_sorted_by_date(){
+
+        when(postRepository.findByOrderByCreationDate()).thenReturn(posts);
+        List<Post> actualPosts = postService.findByOrderByCreationDate();
+        Collections.sort(posts,Collections.reverseOrder());
```
Comment (Yarulika) [inline src/test/java/com/sda/microblogging/service/PostServiceTest.java:71]: None of sorting-related checks can be done here with mocked postRepository as sorting is done there, in postRepository

File: src/test/java/com/sda/microblogging/service/PostServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        verify(postRepository,times(1)).findByOrderByCreationDate();
+    }
+
+    @Test
+    public void get_posts_by_userId() {
+        when(postRepository.findPostsByOwner(1)).thenReturn(posts);
+
+        List<Post> postList = postService.findPostsByOwner(1);
+
+        assertEquals("muhamet",postList.get(1).getOwner().getUsername());
```
Comment (Yarulika) [inline src/test/java/com/sda/microblogging/service/PostServiceTest.java:83]: what is the business function? if user has a few posts - return all of them, I guess. 
In this test looks like: if user has a few posts - return first and make sure it has correct username in owner
Probably check that actually returned postList size = 2 at least, not sure if smth else?

File: src/test/java/com/sda/microblogging/service/PostServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    @Test(expected = NullPointerException.class)
```
Comment (Yarulika) [inline src/test/java/com/sda/microblogging/service/PostServiceTest.java:86]: Let's test exceptions with assertThrows

File: src/test/java/com/sda/microblogging/service/PostServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        verify(postRepository,times(1)).findNumberOfSharesOfPost(1);
+    }
+
+    @Test
+    public void findPostById_return_post(){
+        when(postRepository.findById(any())).thenReturn(Optional.of(posts.get(1)));
+
+        Optional<Post> actualPost = postService.findPostById(5);
+
+        assertEquals(actualPost.get(),posts.get(1));
```
Comment (Yarulika) [inline src/test/java/com/sda/microblogging/service/PostServiceTest.java:111]: assertEquals(): first expected, second - actual

**Review by Yarulika — CHANGES_REQUESTED** [review]
Very good, but might be improved a bit

**Review by bkrmadtya — APPROVED** [review]

**Pull request merged**